common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -111,14 +111,13 @@ public static CalendarInterval fromCaseInsensitiveString(String s) {
     }
 
     long months = toLong(m.group(1)) * 12 + toLong(m.group(2));
-    long microseconds = toLong(m.group(3)) * MICROS_PER_WEEK;
-    microseconds += toLong(m.group(4)) * MICROS_PER_DAY;
-    microseconds += toLong(m.group(5)) * MICROS_PER_HOUR;
+    long days = toLong(m.group(3)) * 7 + toLong(m.group(4));
+    long microseconds = toLong(m.group(5)) * MICROS_PER_HOUR;
     microseconds += toLong(m.group(6)) * MICROS_PER_MINUTE;
     microseconds += toLong(m.group(7)) * MICROS_PER_SECOND;
     microseconds += toLong(m.group(8)) * MICROS_PER_MILLI;
     microseconds += toLong(m.group(9));
-    return new CalendarInterval((int) months, microseconds);
+    return new CalendarInterval((int) months, (int) days, microseconds);
```

Comment:
I think it is better to catch int overflow by `Math.toIntExact` instead of silently ignore it.

Reply:
or maybe we can change this to long

Reply:
Valid range of dates/timestamps is [0..10000). So, we should support days up to `10000 * 366 = 3660000`. Even `int` is so much but it is ok.